django_celery_beat/models.py

```diff
@@ -314,21 +314,26 @@ class Meta:
 
     def validate_unique(self, *args, **kwargs):
         super(PeriodicTask, self).validate_unique(*args, **kwargs)
-        if not self.interval and not self.crontab and not self.solar:
+        schedule_info = {'interval': self.interval,
+                         'crontab': self.crontab,
+                         'solar': self.solar}
+        selected_schedule_types = [schedule_type
+                                   for schedule_type in schedule_info
+                                   if schedule_info[schedule_type]]
+
+        if len(selected_schedule_types) == 0:
```

Comment:
could you explain the logic of if len(selected_....) == 0: ?

Reply:
`len(selected_schedule_types) == 0` means that no schedule type is selected , and there should be one and only one schedule type selected.